### PR TITLE
Add "module" template to new-module and -project wizards in Android Studio

### PR DESCRIPTION
**Pull request by stevemessick**
@pq @devoncarew 

This is behind an experimental flag so it won't affect anyone yet. If anyone is using the feature they might notice that Flutter-specific entries no longer show up in the New Module wizard when it is run from an Android project. It didn't make sense for them to be included, anyway.

WIP: generated project does not open correctly.

Reformatting moved some code around.

### CODE REVIEW

**Comment by stevemessick** [comment]
There's a build failure. That's not the fault of this code, but looks like a problem with the plugin tool. For some unknown reason the edits it performs for the 2017.3 build are not working after editing one of the affected classes. I'll look into it, but you can review this code as-is.

File: flutter-studio/src/io/flutter/module/FlutterDescriptionProvider.java [hunk 1]
```diff
@@ -25,20 +30,39 @@
...
+        }
+      }
+    }
+    if (projectHasFlutter) {
+      // Makes no sense to add Flutter templates to Android projects...
+      res.add(new FlutterApplicationGalleryEntry(sharedModel));
+      res.add(new FlutterPluginGalleryEntry(sharedModel));
+      res.add(new FlutterPackageGalleryEntry(sharedModel));
+    }
+    if (System.getProperty("flutter.experimental.modules", null) != null) {
```
Comment (devoncarew) [inline flutter-studio/src/io/flutter/module/FlutterDescriptionProvider.java]: Let's definitely plan on moving this option into user preferences (in future PRs).

**Comment by stevemessick** [comment]
Sounds good. I'm going to hold this for a bit while I look into the build failure. No point merging this if it breaks everything.

**Comment by stevemessick** [comment]
#2489 makes the build green. Merging now.

**Pull request merged**